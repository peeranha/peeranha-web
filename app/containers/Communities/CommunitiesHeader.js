import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import * as routes from 'routes-config';

import commonMessages from 'common-messages';
import createCommunityIcon from 'svg/createCommunity';

import Base from 'components/Base';
import NavigationButton from 'components/Button/Contained/Navigation';
import A from 'components/A';
import Icon from 'components/Icon';

import messages from './messages';

const suggestedCommunitiesRoute = routes.suggestedCommunities();
const communitiesRoute = routes.communities();

const CommunitiesHeader = /* istanbul ignore next */ ({
  goToCreateCommunityScreen,
  SubHeader,
  changeSorting,
  sorting,
  communitiesNumber,
}) => {
  const path = window.location.pathname + window.location.hash;

  return (
    <div>
      <Base
        className="d-flex align-items-center justify-content-between"
        position="top"
      >
        <div>
          <A to={communitiesRoute} href={communitiesRoute}>
            <NavigationButton isLink={path !== communitiesRoute}>
              <FormattedMessage {...commonMessages.communities} />
            </NavigationButton>
          </A>

          <A to={suggestedCommunitiesRoute} href={suggestedCommunitiesRoute}>
            <NavigationButton isLink={path !== suggestedCommunitiesRoute}>
              <FormattedMessage {...messages.voting} />
            </NavigationButton>
          </A>
        </div>

        <div>
          <NavigationButton
            onClick={goToCreateCommunityScreen}
            className="d-inline-flex align-items-center p-0"
            isLink
          >
            <Icon icon={createCommunityIcon} />
            <FormattedMessage {...messages.suggestCommunity} />
          </NavigationButton>
        </div>
      </Base>

      <Base position="bottom">
        <SubHeader
          changeSorting={changeSorting}
          sorting={sorting}
          communitiesNumber={communitiesNumber}
        />
      </Base>
    </div>
  );
};

CommunitiesHeader.propTypes = {
  goToCreateCommunityScreen: PropTypes.func,
  SubHeader: PropTypes.any,
  changeSorting: PropTypes.func,
  sorting: PropTypes.object,
  communitiesNumber: PropTypes.number,
};

export default CommunitiesHeader;
