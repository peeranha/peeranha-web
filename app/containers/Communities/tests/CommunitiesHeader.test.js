import CommunitiesHeader from '../CommunitiesHeader';

const props = {
  goToCreateCommunityScreen: jest.fn(),
};

describe('CommunitiesHeader', () => {
  it('test', () => {
    expect(CommunitiesHeader(props)).toMatchSnapshot();
  });
});
