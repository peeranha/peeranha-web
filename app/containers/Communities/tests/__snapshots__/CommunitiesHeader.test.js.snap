// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunitiesHeader test 1`] = `
<div>
  <Base
    className="d-flex align-items-center justify-content-between"
    position="top"
  >
    <div>
      <A
        href="/communities"
        to="/communities"
      >
        <Span
          isLink={true}
        >
          <FormattedMessage
            id="app.common.communities"
            values={Object {}}
          />
        </Span>
      </A>
      <A
        href="/communities/suggested"
        to="/communities/suggested"
      >
        <Span
          isLink={true}
        >
          <FormattedMessage
            id="app.containers.Communities.voting"
            values={Object {}}
          />
        </Span>
      </A>
    </div>
    <div>
      <Span
        className="d-inline-flex align-items-center p-0"
        isLink={true}
        onClick={[MockFunction]}
      >
        <UNDEFINED
          icon="
  <svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 18 18\\">
      <g fill=\\"#7699FF\\" fill-opacity=\\".2\\" fill-rule=\\"nonzero\\" stroke=\\"#576FED\\">
          <circle cx=\\"4.5\\" cy=\\"4.5\\" r=\\"4\\" transform=\\"translate(5)\\"/>
          <circle cx=\\"4.5\\" cy=\\"4.5\\" r=\\"4\\" transform=\\"translate(1 7)\\"/>
          <circle cx=\\"4.5\\" cy=\\"4.5\\" r=\\"4\\" transform=\\"translate(9 7)\\"/>
      </g>
  </svg>
"
        />
        <FormattedMessage
          id="app.containers.Communities.suggestCommunity"
          values={Object {}}
        />
      </Span>
    </div>
  </Base>
  <Base
    position="bottom"
  >
    <UNDEFINED />
  </Base>
</div>
`;
